joomla/joomla-cms
# JForm add field option

Currently we are already able to add remove form fields from JForm, update field attributes in the preprocessForm triggers. One thing that has always bothered me is the inability to add new options the the list type form fields.

What I propose here might not be the best solution, however this can be a good point to start with.

---
### Summary of Changes

Allow adding a new list option to `JForm` fields.
Allow additional attributes to the option element created.
Allow adding `<optgroup>` groups in the list options. (will be used only if form field class cares for)

### Testing Instructions
* Create a new `JForm` instance with a list type form field.
* Call `$form->addFieldOption('fieldname', 'value', 'Option Text');` on the `JForm` instance.
* See the output list options.

### Expected result
The new option should be visible in the list dropdown.

### Documentation Changes Required
Update JForm documentation.
